app/models/katello/content_view_deb_filter.rb

```diff
@@ -0,0 +1,59 @@
+module Katello
+  class ContentViewDebFilter < ContentViewFilter
+    CONTENT_TYPE = Deb::CONTENT_TYPE
+
+    has_many :deb_rules, :dependent => :destroy, :foreign_key => :content_view_filter_id,
+                                   :class_name => "Katello::ContentViewDebFilterRule"
+    validates_lengths_from_database
+
+    def generate_clauses(repo)
+      package_filenames = []
+
+      self.deb_rules.each do |rule|
+        package_filenames.concat(query_debs(repo, rule))
+      end
+
+      if self.original_packages
+        package_filenames.concat(repo.packages_without_errata.map(&:filename))
```

Comment:
Does this method make sense for Deb? Is it being used anywhere?

Reply:
Yeah it's not needed here as we deal with errata separately for debian so I removed this condition as well as function definition.